mcp/tools.go

```diff
@@ -58,6 +58,21 @@ type CallToolRequest struct {
 	} `json:"params"`
 }
 
+// GetArguments returns the Arguments as map[string]any for backward compatibility
+// If Arguments is not a map, it returns an empty map
+func (r CallToolRequest) GetArguments() map[string]any {
+	if args, ok := r.Params.Arguments.(map[string]any); ok {
+		return args
+	}
+	return map[string]any{}
+}
+
+// GetRawArguments returns the Arguments as-is without type conversion
+// This allows users to access the raw arguments in any format
+func (r CallToolRequest) GetRawArguments() any {
+	return r.Params.Arguments
+}
+
```

Comment:
We should also consider adding a `BindArguments` function to unmarshal the arguments into a struct as suggested in the issue.

We could also add some common helper functions like `GetString` or `RequireInt` or `GetStringSlice` as well.